lib/developer_portal/app/controllers/developer_portal/admin/messages/trash_controller.rb

```diff
@@ -26,13 +28,13 @@ def destroy
   def empty
     current_account.hidden_messages.destroy_all
 
-    flash[:notice] = 'The trash was emptied.'
+    flash[:notice] = 'Received messages have been deleted.'
```

Comment:
If we destroy sent messages that were deleted (included in `#trashed_messages`), they will also disappear from the inbox of the provider, which is VERY weird - what was sent should not be deleted.

Deleting MessageRecipient (which `#hidden_messages` returns) is fine, because we only delete the part of the message that "belongs" to the developer account.

But this means that clicking "Empty trash" will only delete the received messages, and not the sent ones.

By the way, on the admin portal we don't allow deleting messages in trash at all.

Reply:
but this will look weird, saying Received messages are deleted.

technically you are right, but empty trash we expect that all messages will be deleted either its sent or received so yes Deleting MessageRecipient  is good so message will stay on provider side, but it must be deleted from developer portal.

Reply:
Alternatively, we could just remove the "Empty trash" button... 
But probably people might complain :grimacing:

Reply:
yeah I mean that's definitely a bad idea removing "Empty trash" button , but yes messages remains in trash we need to remove them also

Reply:
Well, what I meant is that we can't delete sent messages from trash... I mean, we technically can, but then they will disappear forever, including from the Inbox of the provider. I think this should not happen.

Imagine - you received an email in your Inbox. Then after some time this email just disappears without you doing anything! Confusing? Yes!